central/signatureintegration/datastore/datastore_impl.go

```diff
@@ -149,6 +158,45 @@ func (d *datastoreImpl) verifyIntegrationIDDoesNotExist(ctx context.Context, id
 	return nil
 }
 
+func (d *datastoreImpl) verifyIntegrationIDIsNotInPolicy(ctx context.Context, id string) error {
+	// We want to avoid deleting a signature integration which is referenced by any policy. If that is the case,
+	// stop the deletion of the policy and return an appropriate error message to the user.
+	policies, err := d.policyStore.GetAllPolicies(ctx)
+	if err != nil {
+		return errox.InvariantViolation.CausedBy(err.Error())
+	}
+
+	var policiesContainingID []string
+	for _, p := range policies {
+		if checkIfPolicyContainsID(id, p) {
+			policiesContainingID = append(policiesContainingID, p.GetName())
+		}
+	}
+
+	if len(policiesContainingID) != 0 {
+		return errox.ReferencedByAnotherObject.Newf("cannot delete signature integration %q since the "+
+			"following policies are referencing it: [%s]", id, strings.Join(policiesContainingID, ","))
```

Comment:
nit: maybe it would be better to include signature integration name here

Reply:
The golden rule I strive to use: do not include anything that is known to the caller. This helps avoid error messages like `DELETE for resource X failed: cannot remove X: cannot delete X because Y is zyzzzzing`.